src/wallet/wallet.cpp

```diff
@@ -1644,7 +1644,7 @@ bool CWallet::AddToWallet(const CWalletTx& wtxIn, bool fFromLoadWallet, CWalletD
         wtx.MarkDirty();
 
         // Notify UI of new or updated transaction
-        NotifyTransactionChanged(this, hash, fInsertedNew ? CT_NEW : CT_UPDATED);
+        NotifyTransactionChanged(hash);
```

Comment:
We call `AddToWallet` from `CommitTransaction`, so we can determine at this point whether the transaction was generated by us, or received from some external location (the mempool, or `sendrawtransaction`). I think that would be useful information to pass through somehow, so we can split `TX_GENERATED` up correctly.